backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/util/git/GitService.java

```diff
@@ -97,6 +97,31 @@ public static Set<URI> getVCSChangedFiles(@Nullable Path baseDir) {
     }
   }
 
+  /**
+   * Retrieves the Git remote URL for the origin remote from the repository.
+   *
+   * @param baseDir the base directory of the project
+   * @return Optional containing the remote URL if found, empty otherwise
+   */
+  public static String getRemoteUrl(@Nullable Path baseDir) {
```

Comment:
We usually mark methods returning null with `@CheckForNull`. This helps to document what can be returned, and it also helps for linters to detect null dereferences.